Integrations/src/main/java/io/deephaven/integrations/learn/Future.java

```diff
@@ -47,17 +49,16 @@
     public Object get() {
 
         if (!called) {
-            try {
+            try (final RowSet rowSet = makeRowSet()) {
                 Object[] gathered = new Object[inputs.length];
 
                 for (int i = 0; i < inputs.length; i++) {
-                    gathered[i] = gather(inputs[i], colSets[i]);
+                    gathered[i] = gather(inputs[i], colSets[i], rowSet);
                 }
 
                 result = func.apply(gathered);
             } finally {
-                rowSet.close();
-                rowSet = null;
+                rowSetBuilder = null;
```

Comment:
I would do this inside of `makeRowSet()`. I also would probably use `rowSetBuilder == null` instead of `rowSetBuilt` (and get rid of the extra boolean).